pkg/distributor/distributor.go

```diff
@@ -904,42 +904,46 @@ func extractLogLevelFromLogLine(log string) string {
 	}
 
 	if firstNonSpaceChar == '{' && lastNonSpaceChar == '}' {
-		if strings.Contains(log, `:"err"`) || strings.Contains(log, `:"ERR"`) ||
-			strings.Contains(log, `:"error"`) || strings.Contains(log, `:"ERROR"`) {
+		levelIndex := strings.Index(log, `"level":`)
```

Comment:
problem is that it could be Level, LEVEL, serverity, Serverity.

Reply:
yeah, i've mentioned it in the caveats in notes.. can we impose this in our documentation? or make this string configurable (case sensetive)?

Reply:
what if we pass a hint to this function? for example, I think we could change [this code](https://github.com/grafana/loki/blob/ffe684c330bcd65f9b07a02d6f93bb475106becc/pkg/distributor/distributor.go#L378) to something like the following

```
firstEntry := stream.Entries[0]
firtsStructuredMetadata := logproto.FromLabelAdaptersToLabels(firstEntry.StructuredMetadata)
prevTs := firstEntry.Timestamp
addLogLevel := validationContext.allowStructuredMetadata &&
				validationContext.discoverLogLevels &&
				!lbs.Has(labelLevel)

logLevelField := detectLogLevelFieldFromLogEntry(firstEntry, firtsStructuredMetadata)
```

this would allow `detectLogLEvelFieldFromLogEntry` to do a case insensitive search for a bunch of different options (ie. `level`, `severity`, etc) since it was only doing it once for the whole stream per push. We could then pass that as a hint to `detectLogLevelFromLogEntry`, allowing it to short-circuit when passed.